CUBRID/cubrid
# [LETS-10] Do not persist info about temporary volumes

http://jira.cubrid.org/browse/LETS-10

The creation of temporary data volumes is not meant to be replicated on page server. Each server has its own temporary data.
As a consequence, temporary data creation cannot be logged and information on temporary data cannot be saved to disk.

Implementation:
- removed temporary volumes info from `boot_dbparm` struct and replaced with dummy same-alignment dummy data in order to maintain backwards compatibility
- added a new struct `boot_temp_info` which is used to store info about temporary volumes and is used to remove temporary volumes upon server clean-close
- in case of server abnormal termination - and when temporary volumes are present - the function `boot_remove_unknown_temp_volumes` has been adapted to work without any additional data: it checks for the presence of temporary volumes starting from `LOG_MAX_DBVOLID` downwards and removes them in ascending order
- adapted function `boot_dbparm_save_volume`  to only log undo data for other than `DB_TEMPORARY_VOLTYPE` volume type

Manual tests:
- connect with `csql` console with user `dba` to a running database instance (preferably one where the volume size is small, eg: `cubrid createdb sdb --db-volume-size=20m --log-volume-size=20m en_US`)
- execute, in at least one or more consoles, statements that lead to creation of temporary volumes:
```
select 'a very long string' from _db_class a, _db_class b, _db_class c, _db_class d;
```
- check that temporary volumes are being created; be sure that every time the statement is executed, a different string is used (to prevent results reuse); the more consoles such a statement is executed in, the more temporary volume files are created
- close the server normally, observe that temporary volume files are removed
- repeat the test but kill the `cub_server` process while temporary volume files are still on disk, restart the server and observe that the temporary volume files are removed upon start

Automatic tests:
- a full suite of shell tests has been executed
